clouddrift/kinematics.py

```diff
@@ -3,9 +3,256 @@
+    )
+    cor_freq_max = np.max(cor_freq * 1.05)
+    cor_freq_min = np.max(
+        [np.min(cor_freq * 0.95), 2 * np.pi / (time_step * data_length)]
+    )
+
+    # logarithmically distributed frequencies for wavelet analysis
+    radian_frequency = morse_logspace_freq(
+        gamma,
+        beta,
+        data_length,
+        (0.05, cor_freq_max * time_step),
+        (5, cor_freq_min * time_step),
+        density,
+    )  # frequencies in radian per unit time
+
+    # wavelet transform on a sphere
+    # unwrap longitude recasted in [0,360)
+    longitude_unwrapped = np.unwrap(recast_lon360(longitude), period=360)
+
+    # convert lat/lon to Cartesian coordinates x, y , z
+    x, y, z = spherical_to_cartesian(longitude_unwrapped, latitude)
+
+    # wavelet transform of x, y, z
+    wavelet, _ = morse_wavelet(data_length, gamma, beta, radian_frequency)
+    wx = wavelet_transform(x, wavelet, boundary="periodic")
+    wy = wavelet_transform(y, wavelet, boundary="periodic")
+    wz = wavelet_transform(z, wavelet, boundary="periodic")
+
+    longitude_new, latitude_new = cartesian_to_spherical(
+        x - np.real(wx), y - np.real(wy), z - np.real(wz)
+    )
+
+    # convert transforms to horizontal displacements on tangent plane
+    wxh, wyh = cartesian_to_tangentplane(wx, wy, wz, longitude_new, latitude_new)
+
+    # rotary wavelet transforms to select inertial component; need to divide by sqrt(2)
+    wp = (wxh + 1j * wyh) / np.sqrt(2)
+    wn = (wxh - 1j * wyh) / np.sqrt(2)
+
+    # find the values of radian_frequency/dt that most closely match cor_freq
+    frequency_bins = [
+        np.argmin(np.abs(cor_freq[i] - radian_frequency / time_step))
+        for i in range(data_length)
+    ]
+
+    # get the transform at the inertial and "anti-inertial" frequencies
+    # extract the values of wp and wn at the calculated index as a function of time
+    # positive is anticyclonic (inertial) in the southern hemisphere
+    # negative is anticyclonic (inertial) in the northern hemisphere
+    wp = wp[frequency_bins, np.arange(0, data_length)]
+    wn = wn[frequency_bins, np.arange(0, data_length)]
+
+    # index of northen latitude points and index of
+    # southern latitude points
+    north = latitude >= 0
+
+    # initialize the zonal and meridional components of inertial displacements
+    wxhat = np.zeros_like(latitude, dtype=np.complex64)
+    wyhat = np.zeros_like(latitude, dtype=np.complex64)
+    # equations are x+ = 0.5*(z+ + z-) and y+ = -0.5*1j*(z+ - z-)
+    if any(north):
+        wxhat[north] = wn[north] / np.sqrt(2)
+        wyhat[north] = 1j * wn[north] / np.sqrt(2)
+    if any(~north):
+        wxhat[~north] = wp[~north] / np.sqrt(2)
+        wyhat[~north] = -1j * wp[~north] / np.sqrt(2)
+
+    # inertial displacement in meters
+    xhat = np.real(wxhat)
+    yhat = np.real(wyhat)
+
+    return xhat, yhat
+
+
+def residual_positions_from_displacements(
+    longitude: Union[float, np.ndarray],
```

Comment:
Here you put `float` or `nd.array`, but the first line validates the shape:
```
if not longitude.shape == latitude.shape == x.shape == y.shape:
```
which doesn't exist for a `float`.

Reply:
Yes, I removed this check. The tests pass now.